env0/resource_configuration_variable.go

```diff
@@ -51,18 +51,21 @@ func resourceConfigurationVariable() *schema.Resource {
 				Description:   "create the variable under this project, not globally",
 				Optional:      true,
 				ConflictsWith: []string{"template_id", "environment_id"},
+				ForceNew:      true,
 			},
 			"template_id": {
 				Type:          schema.TypeString,
 				Description:   "create the variable under this template, not globally",
 				Optional:      true,
 				ConflictsWith: []string{"project_id", "environment_id"},
+				ForceNew:      true,
 			},
 			"environment_id": {
 				Type:          schema.TypeString,
 				Description:   "create the variable under this environment, not globally",
 				Optional:      true,
 				ConflictsWith: []string{"template_id", "project_id", "is_required", "is_read_only"},
+				ForceNew:      true,
```

Comment:
Please also apply this to the `type` attribute of the schema. API doesn't support changing an existing variable's type

Reply:
added.